plugins/modules/cloud/misc/proxmox_storage_info.py

```diff
@@ -0,0 +1,190 @@
+#!/usr/bin/python
+# -*- coding: utf-8 -*-
+#
+# Copyright: Tristan Le Guern (@Aversiste) <tleguern at bouledef.eu>
+# GNU General Public License v3.0+ (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
+
+from __future__ import absolute_import, division, print_function
+__metaclass__ = type
+
+
+DOCUMENTATION = '''
+---
+module: proxmox_storage_info
+short_description: Retrieve information about one or more Proxmox VE storages
+version_added: 2.2.0
+description:
+  - Retrieve information about one or more Proxmox VE storages.
+options:
+  storage:
+    description:
+      - Only return informations on a specific storage.
+    aliases: ['name']
+    type: str
+  type:
+    description:
+      - Filter on a specifc storage type.
+    type: str
+author: Tristan Le Guern (@Aversiste)
+extends_documentation_fragment: community.general.proxmox.documentation
+notes:
+  - Storage specific options can be returned by this module, please look at the documentation at U(https://pve.proxmox.com/wiki/Storage).
+'''
+
+
+EXAMPLES = '''
+- name: List existing storages
+  community.general.proxmox_storage_info:
+    api_host: helldorado
+    api_user: root@pam
+    api_password: "{{ password | default(omit) }}"
+    api_token_id: "{{ token_id | default(omit) }}"
+    api_token_secret: "{{ token_secret | default(omit) }}"
+  register: proxmox_storages
+
+- name: List NFS storages only
+  community.general.proxmox_storage_info:
+    api_host: helldorado
+    api_user: root@pam
+    api_password: "{{ password | default(omit) }}"
+    api_token_id: "{{ token_id | default(omit) }}"
+    api_token_secret: "{{ token_secret | default(omit) }}"
+    type: nfs
+  register: proxmox_storages_nfs
+
+- name: Retrieve information about the lvm2 storage
+  community.general.proxmox_storage_info:
+    api_host: helldorado
+    api_user: root@pam
+    api_password: "{{ password | default(omit) }}"
+    api_token_id: "{{ token_id | default(omit) }}"
+    api_token_secret: "{{ token_secret | default(omit) }}"
+    storage: lvm2
+  register: proxmox_storage_lvm
+'''
+
+
+RETURN = '''
+proxmox_storage:
```

Comment:
One thing you might want to think about is whether this name should be shortened or not. Having something shorter like `storages` makes using result variables `proxmox_storage_lvm` simpler, since you'd have `proxmox_storage_lvm.storages` instead of `proxmox_storage_lvm.proxmox_storage`. On the other hand, if users use very short variables, like `x`, then `x.proxmox_storage` is more descriptive.

I don't have a clear opinion on this, I just want to point this out so that you (and others) can think about it :)

Reply:
I quickly checked what other info modules did: `vultr_account_info` from Vultr collection returns a `vultr_account_info` object, `compute_flavor_info` from OpenStack collection returns a `openstack_flavors` object and `gcp_bigquery_table_info` from GCP collection returns a `resources` object.

There seems to be no consensus on this ?

I personally prefer the naming used in OpenStack modules and thus should rename the return value to the plural form.

Reply:
> There seems to be no consensus on this ?

Yep, it's pretty much wild west ;-)

Reply:
I updated the return value to reflect the reality.